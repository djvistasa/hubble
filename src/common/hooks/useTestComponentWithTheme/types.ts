type IUseTestComponentWithTheme = {};

export type { IUseTestComponentWithTheme };
