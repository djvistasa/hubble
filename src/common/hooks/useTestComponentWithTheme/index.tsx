import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme";

import { RenderResult } from "@testing-library/react";

import { ReactElement } from "react";
import { QueryClient, QueryClientProvider } from "react-query";

function useTestComponentWithTheme() {
  const renderWithTheme = (component: ReactElement<any, any>): RenderResult => {
    const queryClient = new QueryClient();
    return render(
      <QueryClientProvider client={queryClient}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </QueryClientProvider>
    );
  };

  return renderWithTheme;
}

export default useTestComponentWithTheme;
