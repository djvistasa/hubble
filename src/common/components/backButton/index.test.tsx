import useTestComponentWithTheme from "@hooks/useTestComponentWithTheme";
import { describe, expect, it } from "vitest";

import BackButton from ".";

describe("BackButton", () => {
  const mountWithTheme = useTestComponentWithTheme();

  it("should render correctly", () => {
    const component = mountWithTheme(<BackButton />);
    expect(component).toMatchSnapshot();
  });
});
