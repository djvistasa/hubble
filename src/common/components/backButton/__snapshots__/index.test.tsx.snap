// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BackButton > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="styles__StyledBackButton-sc-l4ccil-0 kHXVBS"
      >
        <svg
          fill="black"
          height="20"
          viewBox="0 0 100 100"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 50C18 47.2472 19.0448 44.6003 21.03 42.4828L59.8981 3.09688C63.973 -1.03229 70.66 -1.03229 74.7349 3.09688C78.8098 7.22605 78.8098 13.8962 74.7349 18.0254L43.2852 49.8941L74.9438 81.9746C79.0187 86.1038 79.0187 92.774 74.9438 96.9031C70.869 101.032 64.2865 101.032 60.2116 96.9031L21.03 57.5172C19.0448 55.3997 18 52.7528 18 50Z"
            fill="#989099"
          />
        </svg>
        <p>
          Back
        </p>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="styles__StyledBackButton-sc-l4ccil-0 kHXVBS"
    >
      <svg
        fill="black"
        height="20"
        viewBox="0 0 100 100"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18 50C18 47.2472 19.0448 44.6003 21.03 42.4828L59.8981 3.09688C63.973 -1.03229 70.66 -1.03229 74.7349 3.09688C78.8098 7.22605 78.8098 13.8962 74.7349 18.0254L43.2852 49.8941L74.9438 81.9746C79.0187 86.1038 79.0187 92.774 74.9438 96.9031C70.869 101.032 64.2865 101.032 60.2116 96.9031L21.03 57.5172C19.0448 55.3997 18 52.7528 18 50Z"
          fill="#989099"
        />
      </svg>
      <p>
        Back
      </p>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
